### PR TITLE
Recipe App Version 1.0

### PR DESCRIPTION
**Pull request by ankitt26**
In this Project, We did created a Recipe-app

Created-
- Login page and registration page:
    - Should be built with Devise.
- Food list (**only for a group that has less than 3 members**):
    - Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - Should display a toggle button that allows for a recipe to be made public or private.
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- Write unit and integration tests
- Inventories list:
    - Should display a list of inventories created by the logged-in user as in the wireframe.
    - Should lead to inventory details.
    - If the user is the owner of the inventory, should allow to delete it.
- Inventory details:
    - Should display the inventory details as in the wireframe.
    - Should lead to a form that allows users to add new food.
- Recipe details:
    - Should have a modal to choose an inventory to create a shopping list with, as in the wireframe.
- Inventory shopping list:
    - Should show the list of food that is missing by comparing the food in the recipe with the food in the inventory.
    - Should count the total food items and total price of the missing food.

ER diagram - 
recipe_erd

### CODE REVIEW

**Comment by ankitt26** [comment]
_Note: Please reload the recipe_details page when you Add Food or Generate Shopping List. This is because the Bootstrap CDN loads slowly._

File: spec/models/inventory_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Inventory, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (elisha2kyakpo1) [inline spec/models/inventory_spec.rb]: - [ ] Kindly test this model as well. All the models should be tested.

File: spec/models/recipe_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Recipe, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (elisha2kyakpo1) [inline spec/models/recipe_spec.rb]: - [ ]  Kindly test this model as well. All the models should be tested.

File: app/models/recipe_food.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def value
+    return 'no food error' unless food
+
+    food.price * quantity
+  end
+
+  def self.total_price(recipe_foods)
+    recipe_foods.reduce(0) { |sum, recipe_food| sum + recipe_food.value }
+  end
```
Comment (elisha2kyakpo1) [inline app/models/recipe_food.rb]: - [ ] This model should be tested, do not forget to test the total price function.

File: app/models/recipe.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  has_many :recipe_foods, dependent: :destroy
+  has_many :foods, through: :recipe_foods
+
+  def foods_in_recipe
+    foods
+  end
+
+  def food_quantity(food)
+    recipe_food = recipe_foods.find_by(food_id: food.id)
+    recipe_food&.quantity
```
Comment (elisha2kyakpo1) [inline app/models/recipe.rb:12]: - [ ] Kindly add tests for this model as well.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @ankitt26, @Reggeditt, @forLoop94, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Beautiful UI👏
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly add tests for all the pending tests and fix the failing ones. We need to test all the models and integration specs for the frontend.


testing


tests


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by laiifuu — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: :tada: 

## Hi @ankitt26  :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

friends-joey

## Highlights
- [x] Changes required by the previous reviewer were implemented :+1: 
- [x] All tests passed :+1: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and :computer: Happy coding :computer: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks @laiifuu  for Approving the Project 😊🎊

**Pull request merged**